.github/workflows/build.yml

```diff
@@ -54,173 +89,134 @@ jobs:
+        BUILDENV_BASEPATH: ${{ github.workspace }}/buildenv
 
-    - name: Deploy artifacts
-      uses: actions/upload-artifact@v2
-      with:
-        name: Ubuntu DEB
-        path: cmake_build/*.deb
-
-  build-mac:
-    name: macOS 10.15
-    runs-on: macos-10.15
-    env:
-      APPLE_CODESIGN_IDENTITY: 2C2B5D3EDCE82BA55E22E9A67F16F8D03E390870
-    steps:
-    - name: clone Git repository
-      uses: actions/checkout@v2
+    - name: "Set up ccache"
+      run: ccache -M 5G
+      env:
+        PATH: ${{ steps.buildenv.outputs.path }}
 
-    - name: ccache
+    - name: "Set up ccache cache"
       uses: actions/cache@v2
       with:
-        path: /Users/runner/Library/Caches/ccache
+        path: ${{ matrix.ccache_path }}
         key: ${{ runner.os }}-ccache-${{ github.head_ref }}-${{ github.run_number }}
         restore-keys: |
           ${{ runner.os }}-ccache-${{ github.head_ref }}
           ${{ runner.os }}-ccache
 
-    - name: read build environment file
-      id: buildenv_name
-      run: tools/macos_buildenv.sh name --ghactions
-
-    - name: macOS build environment cache
-      uses: actions/cache@v2
-      env:
-        cache-name: macOS build environment
-      with:
-        path: /Users/runner/buildenv
-        key: macOS-build-environment-${{ steps.buildenv_name.outputs.name }}
-
-    - name: Import code signing identity
-      env:
-        MACOS_CODESIGN_OPENSSL_PASSWORD: ${{ secrets.MACOS_CODESIGN_OPENSSL_PASSWORD }}
-        MACOS_CODESIGN_CERTIFICATE_PASSWORD: ${{ secrets.MACOS_CODESIGN_CERTIFICATE_PASSWORD }}
-      run: |
-        if [ -n "$MACOS_CODESIGN_OPENSSL_PASSWORD" ] && [ -n "$MACOS_CODESIGN_CERTIFICATE_PASSWORD" ]; then
-          # Decrypt the certificate
-          openssl enc -aes-256-cbc -d -md sha512 \
-            -k ${MACOS_CODESIGN_OPENSSL_PASSWORD} \
-            -in /Users/runner/work/mixxx/mixxx/cmake/macos_developer_id_codesign_certificate.p12.enc \
-            -out ~/certificate.p12
-
-          # Create a temporary keychain for the certificate and import it.
-          security create-keychain -p mixxx Mixxx.keychain
-          security unlock-keychain -p mixxx Mixxx.keychain
-          security import ~/certificate.p12 -k Mixxx.keychain \
-            -P ${MACOS_CODESIGN_CERTIFICATE_PASSWORD} -T /usr/bin/codesign
-          security set-key-partition-list -S apple-tool:,apple: -k mixxx Mixxx.keychain
-          # Add keychain to search list
-          security list-keychains -s Mixxx.keychain
-
-          CMAKE_ARGS="-DAPPLE_CODESIGN_IDENTITY=${APPLE_CODESIGN_IDENTITY}"
-        fi
-        echo "::set-output name=cmake_args::${CMAKE_ARGS}"
-
-    - name: Set up macOS build environment
-      id: buildenv
-      run: tools/macos_buildenv.sh setup --ghactions
-
-    - name: Set up ccache
-      run: ccache -M 5G
-      env:
-        PATH: ${{ steps.buildenv.outputs.path }}
-
-    - name: Create build directory
+    - name: "Create build directory"
       run: mkdir cmake_build
```

Comment:
Does need to be its own step? How about combining it with the Configure step?

Reply:
The problem is that this depends on the shell. I moved this to a separate step to make windows integration later on. If it turns out to be unnecessary, we can still merge it into the configure step later.

Reply:
okay